frontend/src/components/VScrollButton.vue

```diff
@@ -44,8 +44,13 @@ export default defineComponent({
         ? positionWithSidebar
         : positionWithoutSidebar
     )
-    const scrollToTop = () => {
-      window.scrollTo({ top: 0, left: 0, behavior: "smooth" })
+    const scrollToTop = (e: MouseEvent) => {
+      const mainPage = e.currentTarget?.closest("#main-page")
+      if (mainPage) {
+        mainPage.scrollTo({ top: 0, left: 0, behavior: "smooth" })
+      } else {
+        window.scrollTo({ top: 0, left: 0, behavior: "smooth" })
+      }
```

Comment:
Nit: is this any better/clearer? Or is there a better way to consolidate this?
```suggestion
      const element = mainPage ?? window
      element.scrollTo({ top: 0, left: 0, behavior: "smooth" })
```

Reply:
That's definitely clearer! I'll check if TS complains about it locally :)

Reply:
It works :)